### ISSUE TITLE
`incus info` show Host-IP Addresses (or nothing) in the IPV4 column after upgrading to kernel 6.9

### ISSUE THREAD

**Issue by qupfer** [lxc/incus#892]
Hi, a small issue for me. Could not find any info about it (that does not mean, there is none :-D )

# Required information

 * Distribution: archlinux
 * The output of "incus info" or if that fails:
 
``` 
[henning@leipzig ~]$ incus info
config:
  core.https_address: '[::]:8443'
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- network_sriov
- console
- restrict_dev_incus
- migration_pre_copy
- infiniband
- dev_incus_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- dev_incus_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- backup_compression
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- snapshot_schedule_aliases
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
- snapshot_expiry_creation
- network_leases_location
- resources_cpu_socket
- resources_gpu
- resources_numa
- kernel_features
- id_map_current
- event_location
- storage_api_remote_volume_snapshots
- network_nat_address
- container_nic_routes
- cluster_internal_copy
- seccomp_notify
- lxc_features
- container_nic_ipvlan
- network_vlan_sriov
- storage_cephfs
- container_nic_ipfilter
- resources_v2
- container_exec_user_group_cwd
- container_syscall_intercept
- container_disk_shift
- storage_shifted
- resources_infiniband
- daemon_storage
- instances
- image_types
- resources_disk_sata
- clustering_roles
- images_expiry
- resources_network_firmware
- backup_compression_algorithm
- ceph_data_pool_name
- container_syscall_intercept_mount
- compression_squashfs
- container_raw_mount
- container_nic_routed
- container_syscall_intercept_mount_fuse
- container_disk_ceph
- virtual-machines
- image_profiles
- clustering_architecture
- resources_disk_id
- storage_lvm_stripes
- vm_boot_priority
- unix_hotplug_devices
- api_filtering
- instance_nic_network
- clustering_sizing
- firewall_driver
- projects_limits
- container_syscall_intercept_hugetlbfs
- limits_hugepages
- container_nic_routed_gateway
- projects_restrictions
- custom_volume_snapshot_expiry
- volume_snapshot_scheduling
- trust_ca_certificates
- snapshot_disk_usage
- clustering_edit_roles
- container_nic_routed_host_address
- container_nic_ipvlan_gateway
- resources_usb_pci
- resources_cpu_threads_numa
- resources_cpu_core_die
- api_os
- container_nic_routed_host_table
- container_nic_ipvlan_host_table
- container_nic_ipvlan_mode
- resources_system
- images_push_relay
- network_dns_search
- container_nic_routed_limits
- instance_nic_bridged_vlan
- network_state_bond_bridge
- usedby_consistency
- custom_block_volumes
- clustering_failure_domains
- resources_gpu_mdev
- console_vga_type
- projects_limits_disk
- network_type_macvlan
- network_type_sriov
- container_syscall_intercept_bpf_devices
- network_type_ovn
- projects_networks
- projects_networks_restricted_uplinks
- custom_volume_backup
- backup_override_name
- storage_rsync_compression
- network_type_physical
- network_ovn_external_subnets
- network_ovn_nat
- network_ovn_external_routes_remove
- tpm_device_type
- storage_zfs_clone_copy_rebase
- gpu_mdev
- resources_pci_iommu
- resources_network_usb
- resources_disk_address
- network_physical_ovn_ingress_mode
- network_ovn_dhcp
- network_physical_routes_anycast
- projects_limits_instances
- network_state_vlan
- instance_nic_bridged_port_isolation
- instance_bulk_state_change
- network_gvrp
- instance_pool_move
- gpu_sriov
- pci_device_type
- storage_volume_state
- network_acl
- migration_stateful
- disk_state_quota
- storage_ceph_features
- projects_compression
- projects_images_remote_cache_expiry
- certificate_project
- network_ovn_acl
- projects_images_auto_update
- projects_restricted_cluster_target
- images_default_architecture
- network_ovn_acl_defaults
- gpu_mig
- project_usage
- network_bridge_acl
- warnings
- projects_restricted_backups_and_snapshots
- clustering_join_token
- clustering_description
- server_trusted_proxy
- clustering_update_cert
- storage_api_project
- server_instance_driver_operational
- server_supported_storage_drivers
- event_lifecycle_requestor_address
- resources_gpu_usb
- clustering_evacuation
- network_ovn_nat_address
- network_bgp
- network_forward
- custom_volume_refresh
- network_counters_errors_dropped
- metrics
- image_source_project
- clustering_config
- network_peer
- linux_sysctl
- network_dns
- ovn_nic_acceleration
- certificate_self_renewal
- instance_project_move
- storage_volume_project_move
- cloud_init
- network_dns_nat
- database_leader
- instance_all_projects
- clustering_groups
- ceph_rbd_du
- instance_get_full
- qemu_metrics
- gpu_mig_uuid
- event_project
- clustering_evacuation_live
- instance_allow_inconsistent_copy
- network_state_ovn
- storage_volume_api_filtering
- image_restrictions
- storage_zfs_export
- network_dns_records
- storage_zfs_reserve_space
- network_acl_log
- storage_zfs_blocksize
- metrics_cpu_seconds
- instance_snapshot_never
- certificate_token
- instance_nic_routed_neighbor_probe
- event_hub
- agent_nic_config
- projects_restricted_intercept
- metrics_authentication
- images_target_project
- images_all_projects
- cluster_migration_inconsistent_copy
- cluster_ovn_chassis
- container_syscall_intercept_sched_setscheduler
- storage_lvm_thinpool_metadata_size
- storage_volume_state_total
- instance_file_head
- instances_nic_host_name
- image_copy_profile
- container_syscall_intercept_sysinfo
- clustering_evacuation_mode
- resources_pci_vpd
- qemu_raw_conf
- storage_cephfs_fscache
- network_load_balancer
- vsock_api
- instance_ready_state
- network_bgp_holdtime
- storage_volumes_all_projects
- metrics_memory_oom_total
- storage_buckets
- storage_buckets_create_credentials
- metrics_cpu_effective_total
- projects_networks_restricted_access
- storage_buckets_local
- loki
- acme
- internal_metrics
- cluster_join_token_expiry
- remote_token_expiry
- init_preseed
- storage_volumes_created_at
- cpu_hotplug
- projects_networks_zones
- network_txqueuelen
- cluster_member_state
- instances_placement_scriptlet
- storage_pool_source_wipe
- zfs_block_mode
- instance_generation_id
- disk_io_cache
- amd_sev
- storage_pool_loop_resize
- migration_vm_live
- ovn_nic_nesting
- oidc
- network_ovn_l3only
- ovn_nic_acceleration_vdpa
- cluster_healing
- instances_state_total
- auth_user
- security_csm
- instances_rebuild
- numa_cpu_placement
- custom_volume_iso
- network_allocations
- zfs_delegate
- storage_api_remote_volume_snapshot_copy
- operations_get_query_all_projects
- metadata_configuration
- syslog_socket
- event_lifecycle_name_and_project
- instances_nic_limits_priority
- disk_initial_volume_configuration
- operation_wait
- image_restriction_privileged
- cluster_internal_custom_volume_copy
- disk_io_bus
- storage_cephfs_create_missing
- instance_move_config
- ovn_ssl_config
- certificate_description
- disk_io_bus_virtio_blk
- loki_config_instance
- instance_create_start
- clustering_evacuation_stop_options
- boot_host_shutdown_action
- agent_config_drive
- network_state_ovn_lr
- image_template_permissions
- storage_bucket_backup
- storage_lvm_cluster
- shared_custom_block_volumes
- auth_tls_jwt
- oidc_claim
- device_usb_serial
- numa_cpu_balanced
- image_restriction_nesting
- network_integrations
- instance_memory_swap_bytes
- network_bridge_external_create
- network_zones_all_projects
- storage_zfs_vdev
- container_migration_stateful
- profiles_all_projects
- instances_scriptlet_get_instances
- instances_scriptlet_get_cluster_members
- instances_scriptlet_get_project
- network_acl_stateless
- instance_state_started_at
- networks_all_projects
- network_acls_all_projects
- storage_buckets_all_projects
- resources_load
- instance_access
- project_access
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
auth_user_name: henning
auth_user_method: unix
environment:
  addresses:
  - 192.168.178.2:8443
  - '[2a02:810a:980:4a00:ea6a:64ff:fed7:ff38]:8443'
  - 10.0.0.1:8443
  - 10.69.83.1:8443
  - '[fd42:4447:c115:fd5::1]:8443'
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIICFzCCAZygAwIBAgIRAONUC67P0Odzeym8TwI2GtwwCgYIKoZIzj0EAwMwOTEZ
    MBcGA1UEChMQTGludXggQ29udGFpbmVyczEcMBoGA1UEAwwTcm9vdEBsZWlwemln
    LjA3cS5kZTAeFw0yNDAzMTMxMjQ5NTZaFw0zNDAzMTExMjQ5NTZaMDkxGTAXBgNV
    BAoTEExpbnV4IENvbnRhaW5lcnMxHDAaBgNVBAMME3Jvb3RAbGVpcHppZy4wN3Eu
    ZGUwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAThZ+sfCwPLs1ceYekzeV9dgXx2Jdpc
    OHeF7DUXwXV3WsdpMch40V2Eh4zuS44zEmqem+A6MnvI50XbKbDUfuVU/jqenMLf
    aWYXu0BMCaVmtmBTBh+OhStJ/x58lR/n6qKjaDBmMA4GA1UdDwEB/wQEAwIFoDAT
    BgNVHSUEDDAKBggrBgEFBQcDATAMBgNVHRMBAf8EAjAAMDEGA1UdEQQqMCiCDmxl
    aXB6aWcuMDdxLmRlhwR/AAABhxAAAAAAAAAAAAAAAAAAAAABMAoGCCqGSM49BAMD
    A2kAMGYCMQDQVZndGMqSlrRdIM4gCDg8Jrbf4Gof5A13AX/NXt1AIgu9ax54aq5K
    7/F4xr4U02oCMQDF4XJ5kiCpkM94rbCthI/rggDizo8hcboD5fiqi8lrXp8on8sZ
    IUgNpN85tjSaO1Y=
    -----END CERTIFICATE-----
  certificate_fingerprint: e4b819094220d423a2d350279c9d9eac1867f4eb5a6207da57aa392249c31e77
  driver: lxc | qemu
  driver_version: 6.0.0 | 9.0.0
  firewall: nftables
  kernel: Linux
  kernel_architecture: x86_64
  kernel_features:
    idmapped_mounts: "true"
    netnsid_getifaddrs: "true"
    seccomp_listener: "true"
    seccomp_listener_continue: "true"
    uevent_injection: "true"
    unpriv_binfmt: "true"
    unpriv_fscaps: "true"
  kernel_version: 6.9.1-arch1-2
  lxc_features:
    cgroup2: "true"
    core_scheduling: "true"
    devpts_fd: "true"
    idmapped_mounts_v2: "true"
    mount_injection_file: "true"
    network_gateway_device_route: "true"
    network_ipvlan: "true"
    network_l2proxy: "true"
    network_phys_macvlan_mtu: "true"
    network_veth_router: "true"
    pidfd: "true"
    seccomp_allow_deny_syntax: "true"
    seccomp_notify: "true"
    seccomp_proxy_send_notify_fd: "true"
  os_name: Arch Linux
  os_version: ""
  project: default
  server: incus
  server_clustered: false
  server_event_mode: full-mesh
  server_name: leipzig.07q.de
  server_pid: 687
  server_version: "6.1"
  storage: btrfs
  storage_version: 6.8.1
  storage_supported_drivers:
  - name: dir
    version: "1"
    remote: false
  - name: zfs
    version: 2.2.99-504_gefbef9e6cc
    remote: false
  - name: btrfs
    version: 6.8.1
    remote: false
``` 

# Issue description

After upgrading the kernel to 6.9 on my archlinux,
the IPV4 column in `incus list` shows wrong values.

 ``` 
[henning@leipzig ~]$ incus list
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
|     NAME      |  STATE  |         IPV4          |                     IPV6                     |   TYPE    | SNAPSHOTS |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| checkmk       | RUNNING |                       | fd42:4447:c115:fd5:216:3eff:fe2c:3dd9 (eth0) | CONTAINER | 4         |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| guacamole     | RUNNING |                       | fd42:4447:c115:fd5:216:3eff:fe23:9f11 (eth0) | CONTAINER | 4         |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| immich        | RUNNING | 192.168.178.2 (eno2)  | fd42:4447:c115:fd5:216:3eff:fee2:5e94 (eth0) | CONTAINER | 9         |
|               |         | 10.69.83.1 (incusbr0) |                                              |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| nextcloud     | RUNNING |                       | fd42:4447:c115:fd5:216:3eff:fe3b:5753 (eth0) | CONTAINER | 6         |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| overleaf      | STOPPED |                       |                                              | CONTAINER | 2         |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| paperless-ngx | RUNNING | 192.168.178.2 (eno2)  | fd42:4447:c115:fd5:216:3eff:fe2f:25a7 (eth0) | CONTAINER | 10        |
|               |         | 10.69.83.1 (incusbr0) |                                              |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| samba         | RUNNING |                       | fd42:4447:c115:fd5:216:3eff:fe86:28c1 (eth0) | CONTAINER | 10        |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| sip           | RUNNING |                       | fd42:4447:c115:fd5:216:3eff:fe18:414a (eth0) | CONTAINER | 9         |
|               |         |                       | 2a02:810a:980:4a00:216:3eff:fe09:153 (eth1)  |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| softether     | RUNNING | 192.168.178.2 (eno2)  | fd42:4447:c115:fd5:bb:b7ff:fefe:6313 (br0)   | CONTAINER | 10        |
|               |         |                       | fd42:4447:c115:fd5:216:3eff:fec2:1981 (eth0) |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| tdns          | RUNNING |                       | fd42:4447:c115:fd5:216:3eff:fe2d:85a2 (eth0) | CONTAINER | 4         |
|               |         |                       | 2a02:810a:980:4a00:216:3eff:feac:fa5f (eth1) |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| unifi         | RUNNING | 192.168.178.2 (eno2)  | fd42:4447:c115:fd5:216:3eff:fe10:f027 (eth0) | CONTAINER | 4         |
|               |         | 10.69.83.1 (incusbr0) | 2a02:810a:980:4a00:216:3eff:fead:94ab (eth1) |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
| vaultwarden   | RUNNING | 192.168.178.2 (eno2)  | fd42:4447:c115:fd5:216:3eff:febc:3c7f (eth0) | CONTAINER | 10        |
|               |         | 10.0.0.1 (wg0)        |                                              |           |           |
+---------------+---------+-----------------------+----------------------------------------------+-----------+-----------+
```` 

 

# Information to attach

* shown ip addresses are from thoe incus host
 -  192.168.178.2 is the IP of the incus-host
-  10.69.83.1 is the IP of the incusbr0 on incus-host
-  10.0.0.1 is the IP of the wireguard interface on incus-host 
* use todays incus from git: c7d025b92ca7709f8e1bf06926d806789a2f7f14
* happens wih release version too (6.1)

**Comment by abdodz1234** [comment]
same here
os: archlinux
kernel: 6.9

**Comment by stgraber** [comment]
Marking as incomplete as our logic is pretty simple on our end so if getting the wrong values, it's likely due to a kernel issue with the netlink call we make to retrieve network information from the network namespace.

It'd be good if someone could do a kernel git bisect to see what commit broke it.

**Comment by qupfer** [comment]
i will try that 
(I never build an kernel before...so that may be need some time  :-D )

**Comment by stgraber** [comment]
I'm preparing the rollout of 6.9.2 in my own package repository so I'll do a few tests and make sure we have any needed fixes in place before rolling it out to folks.

**Comment by banafshehft** [comment]
I have a similar issue, but it only affects my containers. The IP addresses are not displayed; the IPv4 field is blank. However, the containers are functioning correctly and do have IP addresses.
os: archlinux
kernel: 6.9.2

**Comment by stgraber** [comment]
I've started a bisect for this one as there's nothing obviously wrong in our code and that kernel API should really never show information about a network namespace other than the one we're querying...

**Comment by stgraber** [comment]
```
commit cdb2f80f1c10654efc66c1624f66df2b87eabf06
Author: Eric Dumazet <edumazet@google.com>
Date:   Thu Feb 29 11:40:16 2024 +0000

    inet: use xa_array iterator to implement inet_dump_ifaddr()
    
    1) inet_dump_ifaddr() can can run under RCU protection
       instead of RTNL.
    
    2) properly return 0 at the end of a dump, avoiding an
       an extra recvmsg() system call.
    
    Signed-off-by: Eric Dumazet <edumazet@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
```

That's what `git bisect` found here. The code only touches net/ipv4 so that'd be consistent with only IPv4 being affected. I've reached out to the authors to have it be investigated.

**Comment by qupfer** [comment]
Thanks, I will check (just for me) that I end at the same commit.

**Comment by stgraber** [comment]
@mihalicyn found the problematic part of that change and has a fix that I'm now testing in my kernels.

**Comment by stgraber** [comment]
https://github.com/zabbly/linux/commit/be22d289b91db782234b6350aa3b3f9d4b039c26

I'm closing this issue as it's now been confirmed to be a kernel bug and not something we can do anything about from Incus itself.

**Comment by stgraber** [comment]
For those using my (zabbly) packages. I've pushed that fix in my kernel builds.
Once those complete, I'll update the stable and lts-6.0 Incus branches with the needed other fixes to handle Linux 6.9 then will push everything out.
